### PR TITLE
Added a run() function

### PR DESCRIPTION
**Pull request by VeeBallbach**
Adds a ```run()``` function to probably be useful eventually

### CODE REVIEW

File: autonomous.py [hunk 1]
```diff
@@ -2,6 +2,24 @@
...
+        return self
+
+    def execute(self):
+        pass
+
+    def end(self):
+        pass
+
+    def run(self):
+        self.init()
```
Comment (porglezomp) [inline autonomous.py]: we should put in a nested function that does everything in execute then ends, like
```python
def _execute():
    yield from self.execute()
    self.end()
```
then we can `return _execute()` instead

File: autonomous.py [hunk 2]
```diff
@@ -2,6 +2,24 @@
...
+        return self.execute()
 
 class Timed:
```
Comment (porglezomp) [inline autonomous.py]: The `end()` for `Timed` should call the inner `end()`.

File: robot.py [hunk 3]
```diff
@@ -55,22 +60,28 @@ def robotInit(self):
...
-        self.auton.init()
-        self.auton_exec = self.auton.execute()
+        self.auton = autonomous.drive_and_rotate(self.drivetrain, self.gyro)
 
     def autonomousPeriodic(self):
         try:
-            next(self.auton_exec)
+            next(self.auton)
         except StopIteration:
             # WPILib prints a ton of error messages when the motor has no output
```
Comment (porglezomp) [inline robot.py:127]: We should call `self.auton.end()` here

**Review by porglezomp — CHANGES_REQUESTED** [review]

**Comment by VeeBallbach** [comment]
I changed things, but I don't know if I did it right

**Comment by porglezomp** [comment]
Looks like those changes don't work. Remember to run tests (`python3 robot.py test` (or on Windows,`py -3 robot.py test`) so you can find out if the code works.

**Comment by VeeBallbach** [comment]
It passes the tests now, but I'm not sure how to implement your second and third suggestions

**Review by a2aaron — APPROVED** [review]
This appears to have all the changes done!

**Review by porglezomp — APPROVED** [review]
Okay, looks good

**Pull request merged**